pkg/apis/storage/types.go

```diff
@@ -672,3 +672,50 @@ type CSIStorageCapacityList struct {
 	// Items is the list of CSIStorageCapacity objects.
 	Items []CSIStorageCapacity
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeAttributesClass represents a specification of mutable volume attributes
+// defined by the storage driver. The class can be specified during dynamic provisioning
```

Comment:
I was surprised when I read the field level docs to discover "the parameters field itself is immutable".  Can we document here at the resource level that this resource kind is not "mutable", but that a `PersistentVolumeClaim` can change which `VolumeAttributesClass` is refers to do change the parameters it is using?

Reply:
> I was surprised when I read the field level docs to discover "the parameters field itself is immutable". Can we document here at the resource level that this resource kind is not "mutable", but that a `PersistentVolumeClaim` can change which `VolumeAttributesClass` is refers to do change the parameters it is using?

I think it's hard to list some immutable attributes of a volume which depends on the specific storage provider.

> but that a PersistentVolumeClaim can change which VolumeAttributesClass is refers to do change the parameters it is using?

It's forbidden to changes the parameters of a VolumeAttributesClass object whatever it is used or not in a PVC object. If users want to changes attributes of a volume, they have to define a new VolumeAttributesClass object and use it in the PVC object. In the new VolumeAttributesClass object, users can define parameters of volume as they want.